framework/src/NormalisedSumPDF.cpp

```diff
@@ -293,121 +304,81 @@ bool NormalisedSumPDF::SetPhysicsParameters( ParameterSet * NewParameterSet )
 //Return the integral of the function over the given boundary
 double NormalisedSumPDF::Normalisation( DataPoint* NewDataPoint, PhaseSpaceBoundary * NewBoundary )
 {
-	(void)NewDataPoint;
-	(void)NewBoundary;
-	//The evaluate method already returns a normalised value
+	// Calculate and cache the integrals
+	double firstIntegralResult = firstPDF->Integral( NewDataPoint, NewBoundary ) * firstIntegralCorrection;
+	double secondIntegralResult = secondPDF->Integral( NewDataPoint, NewBoundary ) * secondIntegralCorrection;
+	SetCachedIntegrals({firstIntegralResult,secondIntegralResult}, NewDataPoint, NewBoundary);
```

Comment:
ah, never mind I mis-read the PDF function name.

Hmm... is this always safe?
I'm thinking it's safer to store/cache based upon the Phase-Space which has been stored in `self.integrationBoundary` but it would make sense to cache each combination which is common to `NewBoundary` here and the internal boundary.
I'm wondering if there is ever a problem with changing the phase-space here over the stored phase-space internally? Which is correct. I'm constructing arguments for both and this makes caching tricky...